.github/workflows/lint-specs.yml

```diff
@@ -0,0 +1,95 @@
+name: Spec Linting
+
+on:
+  push:
+    paths:
+      - '**.spec'
+    branches: [main, dev, 1.0*]
+  pull_request:
+    paths:
+      - '**.spec'
+    branches: [main, dev, 1.0*]
+
+jobs:
+  spec-lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checkout the branch of our repo that triggered this action
+      - name: Workflow trigger checkout
+        uses: actions/checkout@v2
+
+      - name: Get base commit for PRs
+        if: ${{ github.event_name == 'pull_request' }}
+        run: |
+          git fetch origin ${{ github.base_ref }}
+          echo "base_sha=$(git rev-parse origin/${{ github.base_ref }})" >> $GITHUB_ENV 
+          echo "Merging ${{ github.sha }} into ${{ github.base_ref }}"
+
+      - name: Get base commit for Pushes
+        if: ${{ github.event_name == 'push' }}
+        run: |
+          git fetch origin ${{ github.event.before }}
+          echo "base_sha=${{ github.event.before }}" >> $GITHUB_ENV 
+          echo "Merging ${{ github.sha }} into ${{ github.event.before }}"
+
+      - name: Get the changed files
+        run: |
+          echo "Files changed: '$(git diff-tree --no-commit-id --name-only -r ${{ env.base_sha }} ${{ github.sha }})'"
+          changed_specs=$(git diff-tree --no-commit-id --name-only -r ${{ env.base_sha }} ${{ github.sha }} | { grep "\.spec$" || test $? = 1; })
```

Comment:
What is the reason behind `test $? = 1;` at the very end? It looks like it would make us succeed even if `grep` found no .spec files being modified, which I guess should never happen considering the `paths` at the beginning of this file.

Reply:
If I recall my reasoning (I wrote the cgmanifest script this filter was taken from)... grep returns an error if it doesn't find any matches, so you need to explicitly catch that return code and ignore it.